src/app/modules/auth/auth.controller.ts

```diff
@@ -182,7 +237,7 @@ export const handleSignout: RequestHandler = async (req, res) => {
     return res.sendStatus(StatusCodes.BAD_REQUEST)
   }
 
-  req.session!.destroy((error) => {
+  req.session?.destroy((error) => {
```

Comment:
can we avoid this check by changing the above `if` statement to `if (!req.session || isEmpty(req.session))`?

Reply:
`isEmpty(undefined)` or `isEmpty(null)` will also return `true`, but I agree that is not very obvious. Will add a second conditional.

(All this is moot though, req.session is guaranteed to exist since `express-session` middleware is used)

Reply:
Yeah, I was thinking we can add the second conditional just so TypeScript cooperates